### ISSUE TITLE
Rubygems 3.4.12 "silence_warnings: uninitialized constant `Gem::DefaultUserInteraction` (`NameError`)"

### ISSUE THREAD

**Issue by tadman** [Shopify/tapioca#1485]
There might be a compatibility issue with Rubygems where `Gem::DefaultUserInteraction` is not automatically loaded, as in:

```shell
$ tapioca dsl

.../.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/tapioca-0.11.5/lib/tapioca.rb:22:in `silence_warnings': uninitialized constant Gem::DefaultUserInteraction (NameError)
```

Likely because of this:

```ruby
require 'rubygems'

Gem::DefaultUserInteraction
# =>  uninitialized constant Gem::DefaultUserInteraction (NameError)
```

For what it's worth, force-requiring it does make that visible:

```ruby
require 'rubygems/user_interaction'
```

Though `silence_warnings` doesn't do this currently, and doesn't check that it's loaded before exercising the `use_ui` method.

This is with Rubygems 3.4.12 and Tapioca v0.11.5 on macOS 13.3.1.

**Comment by KaanOzkan** [comment]
Yes looks like in recent RubyGems it's no longer loaded by default. We can manually load it in tapioca.

However, it's [recommended](https://github.com/Shopify/tapioca#getting-started:~:text=install%20the%20binstub%20for%20Tapioca%20in%20your%20app%27s%20bin/%20folder%2C%20so%20that%20you%20can%20use%20bin/tapioca%20to%20run%20commands%20in%20your%20app) that you use the binstubs tapioca generates instead which resolves the issue as well: `bin/tapioca dsl`.

**Comment by tadman** [comment]
The binstub version does work, so that's good, but it's inconvenient that the globally installed one does not.

Should be an easy fix.

**Comment by JamieMagee** [comment]
If I understand this thread correctly, the fix is to `require 'rubygems/user_interaction'` in `lib/tapioca.rb`. If so, I'll create a small PR to fix this.

**Comment by amomchilov** [comment]
I confirmed that this was solved with the merge of #1650. This bug will be gone in the next release :)
